packages/react-hooks/src/useDebouncedCallback.ts

```diff
@@ -8,15 +8,23 @@ import debounce from 'lodash.debounce';
  * component unmounts.
  * @param callback callback function to debounce
  * @param debounceMs debounce milliseconds
+ * @param options debounce options such as leading or trailing behavior
  * @returns a cancelable, debounced function
  */
 export function useDebouncedCallback<TArgs extends unknown[], TResult>(
   callback: (...args: TArgs) => TResult,
-  debounceMs: number
+  debounceMs: number,
+  options: DebounceSettings = {}
 ): DebouncedFunc<(...args: TArgs) => TResult> {
+  const { leading = false, trailing = true, maxWait } = options;
```

Comment:
Yep I figured that out through trial and error, but good to know the code confirmed my suspicions that they were coercing falsey value to false. I should probably add tests for this

I also made this not require a stable object into the hook because to me it's better to hide that in the hook implementation since it's simple. That way the calls are more clear with `useDebouncedCallback(fn, timeout, { leading: true })`